internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -1032,6 +1035,15 @@ func (c *Coordinator) runLoopIteration(ctx context.Context) {
 
 	case c.heartbeatChan <- struct{}{}:
 
+	case <-c.servicePidUpdate:
```

Comment:
Service PID update is a bit of a misleading title, it implies it has to do with the service runtime specifically, but you are just generically pulling this out of the checkin observed so it isn't limited to that.

Remove service from the name to clarify this.